### PR TITLE
added tagged_timeindex function, and applied clang format

### PR DESCRIPTION
**Pull request by vincentberenz**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

- Added function tagged_timeindex, which returns the index at which the time series has been tagged.

- Applied clang format to the full package (disclaimer: I thought this had already be done in a previous commit. It was not. So I will comment the lines corresponding to the updated code)

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

## How I Tested

I updated the unit tests and made sure they pass

[//]: # "Explain how you tested your changes"

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x ] All new functions/classes are documented and existing documentation is updated according to changes.
- [ x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: include/time_series/interface.hpp [hunk 1]
```diff
@@ -95,6 +96,11 @@ class TimeSeriesInterface
      */
     virtual void tag(const Index &timeindex) = 0;
 
+    /*! \brief returns the index at which the time series has been tagged.
+     * Returns the newest timeindex if the time series has never been tagged. 
+     */
+    virtual Index tagged_timeindex() = 0;
```
Comment (vincentberenz) [inline include/time_series/interface.hpp:102]: updated

File: include/time_series/internal/base.hpp [hunk 2]
```diff
@@ -40,6 +40,7 @@ class TimeSeriesBase : public TimeSeriesInterface<T>
     size_t max_length();
     bool has_changed_since_tag();
     void tag(const Index &timeindex);
+    Index tagged_timeindex();
```
Comment (vincentberenz) [inline include/time_series/internal/base.hpp:43]: updated

File: include/time_series/internal/base.hxx [hunk 3]
```diff
@@ -21,6 +20,14 @@ void TimeSeriesBase<P, T>::tag(const Index& timeindex)
     write_indexes();
 }
 
+template <typename P, typename T>
+Index TimeSeriesBase<P, T>::tagged_timeindex()
```
Comment (vincentberenz) [inline include/time_series/internal/base.hxx:24]: updated

File: tests/basic_unit_tests.cpp [hunk 4]
```diff
@@ -183,6 +183,8 @@ TEST(time_series_ut, tag)
     ts.append(20);
     changed = ts.has_changed_since_tag();
     ASSERT_TRUE(changed);
+    Index tagged_index = ts.tagged_timeindex();
```
Comment (vincentberenz) [inline tests/basic_unit_tests.cpp:186]: updated

**Review by luator — APPROVED** [review]

**Pull request merged**